selene_sdk/targets/_genomic_features.pyx

```diff
@@ -31,7 +31,7 @@ def _fast_get_feature_data(int start,
         feature_end = int(row[2])
         index_start = max(0, feature_start - start)
         index_end = min(feature_end - start, query_length)
-        index_feat = feature_index_dict[row[3]]
+        index_feat = feature_index_dict[row[4]]
```

Comment:
If we want to merge these into Selene in the future, one thing that will be important is backwards compatibility.

Right now, most users of Selene expect `row[3]` to be the target label that is being predicted. So for the addition of the strand column, I think we can do one or the other:
(1) Add the strand col as the fourth column for now (most minimal change, but maybe doesn't make as much sense from BED file column organization standpoint)
(2) Add the strand col as the third col and the targets col as the fourth column (what you have now) but our library needs to know to check---whether through specification of an optional parameter or automatically---the correct column for retrieving strand and targets now.

Just something worth thinking about! We can discuss what is the best way. 

You know your own priorities and schedule for this project best: if it will be too time-consuming to make the changes sufficient for pushing this to Selene within your timeline, you should focus on getting a model trained, and then if you have time, refactoring and testing your code in this pull request to make it ready for merging into Selene.